src/analysis/sensitivity.jl

```diff
@@ -32,6 +32,52 @@ function ks_statistic(ks::ApproximateKSTest)::Float64
     return sqrt(n) * ks.δ
 end
 
+"""
+    _get_factor_spec(model_spec::DataFrame, factors::Vector{String})
+
+Get model spec for specified factors.
+
+# Arguments 
+- model_spec : Model specification, from `model_spec(domain)` or `rs.model_spec`
+- factors : Parameters considered for sensitivity analysis
+"""
+function _get_factor_spec(model_spec::DataFrame, factors::Vector{Symbol})
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    return model_spec[factors_to_assess, foi_attributes]
+end
+
+"""
+    _category_bins(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+
+Get number of binnings for categorical variables.
+
+# Arguments
+- S : Number of bins
+- foi_spec : Model specification for factors of interest
+- foi_cat : Vector of length `foi_cat.fieldname` which contains true where the factor is categorical and false otherwise
```

Comment:
```suggestion
# Arguments
- `S` : Number of bins
- `foi_spec` : Model specification for factors of interest
- `foi_cat` : Vector of length `foi_cat.fieldname` which contains `true` where the factor is categorical and `false` otherwise
```

On `foi_cat`: to me this would mean "factor of interest category", but it holds true/false indicating whether a factor is categorical or not. How about calling this `is_cat` ("is categorical") ?

Reply:
Actually on second thought - this seems over-complicated to me.

Why have both the dataframe and the selector as arguments, when the desired subset could be passed in right away?

```julia
_category_bins(S, foi_spec[is_cat, :])
```

Reply:
Good point!